sklearn/preprocessing/_discretization.py

```diff
@@ -149,6 +157,11 @@ def fit(self, X, y=None):
             raise ValueError("Valid options for 'strategy' are {}. "
                              "Got strategy={!r} instead."
                              .format(valid_strategy, self.strategy))
+        valid_dtype = ('float32', 'float64')
```

Comment:
The use of object rather than string dtypes is preferred I think,
```suggestion
        valid_dtype = [np.float32, np.float64]
```

Reply:
Thanks for the edits @rth. Here, the problem is that when the user passes `'float32'` or `'float64'` as a parameter, a `ValueError` will be raised. Instead, we can allow such parameter to be passed so the end-user need not import numpy externally.

Reply:
> Here, the problem is that when the user passes 'float32' or 'float64' as a parameter, a ValueError will be raised. 

It shouldn't be if validating `np.dtype(self.dtype)` as suggested above, as then both `"float32"` and `np.float32` will return the latter.

Reply:
Got it! I think everything should be resolved by now! Please suggest further edits, if any!